lib/mods_display/fields/field.rb

```diff
@@ -57,5 +57,46 @@ def collapse_fields(display_fields)
     def element_text(element)
       element.xpath('.//text()').to_html.strip
     end
+
+    # used for originInfo date fields, e.g. DateCreated, DateIssued ...
+    def date_fields(date_symbol)
+      return_fields = @values.map do |value|
+        date_values = Stanford::Mods::Imprint.new(value).dates([date_symbol])
+        next unless date_values.present?
+
+        ModsDisplay::Values.new(
+          label: I18n.t("mods_display.#{date_symbol.to_s.underscore}"),
+          values: select_best_date(date_values),
+          field: self
+        )
+      end.compact
+      collapse_fields(return_fields)
+    end
+
+    # used for originInfo dates, e.g. for Imprint, DateCreated, DateIssued, etc.
+    def select_best_date(dates)
```

Comment:
moved this from Imprint class so it could be used by both Imprint and the new date fields